src/app/_mermaid_common/forms/directives/dateinput.dir.js

```diff
@@ -44,7 +44,7 @@ angular
         link: function(scope, element, attr, ngModel) {
           var val;
           var $date_input = element;
-          var dataDateFormat = scope.$parent.$parent.widgetFormat;
+          var dataDateFormat = scope.$parent.widgetFormat;
```

Comment:
@tienhoah does this work everywhere, including any modals that might use the directive?

Reply:
scope.$parent.widgetFormat =>  yy-mm-dd 
scope.$parent.$parent.widgetFormat =>  undefined (this happens in Submit record mostly, this is from M1244 bug I found)